iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/writer/HiveIcebergWriterBase.java

```diff
@@ -77,4 +86,39 @@ protected PartitionKey partition(Record row, int specId) {
     partitionKey.partition(wrapper.wrap(row));
     return partitionKey;
   }
+
+  // use a fanout writer only if enabled and the input is unordered and the table is partitioned
+  static PartitioningWriter<Record, DataWriteResult> newDataWriter(
+      Table table, HiveFileWriterFactory writers, OutputFileFactory files, Context context) {
+
+    FileIO io = table.io();
+    boolean useFanoutWriter = context.useFanoutWriter();
+    long targetFileSize = context.targetDataFileSize();
+
+    if (table.spec().isPartitioned() && useFanoutWriter) {
```

Comment:
Fanout writers should be used only when its partitioned? Is this Iceberg / Spark induced behaviour?

Reply:
yes, see SparkPositionDeltaWrite